python/cudf/cudf/core/column/timedelta.py

```diff
@@ -367,6 +368,35 @@ def median(self, skipna: bool = None) -> pd.Timedelta:
             self.as_numerical.median(skipna=skipna), unit=self.time_unit
         )
 
+    def isin(self, values: Sequence) -> ColumnBase:
+        if cudf.utils.dtypes.is_scalar(values):
+            raise TypeError(
+                "only list-like objects are allowed to be passed "
+                f"to isin(), you passed a [{type(values).__name__}]"
+            )
+
+        lhs = self
+        rhs = None
+
+        try:
+            rhs = cudf.core.column.as_column(values)
+
+            if rhs.dtype.kind in {"f", "i", "u"}:
+                return cudf.core.column.full(len(self), False, dtype="bool")
+
+            rhs = rhs.astype(self.dtype)
+            res = lhs._isin_earlystop(rhs)
+            if res is not None:
+                return res
+        except ValueError:
+            # pandas functionally returns all False when cleansing via
+            # typecasting fails
+            return cudf.core.column.full(len(self), False, dtype="bool")
+
+        res = lhs._obtain_isin_result(rhs)
+
+        return res
```

Comment:
I think this is almost exactly the same function as the one for `DatetimeColumn`. Is it worth sharing the implementation? Or could they be different later somehow?

Reply:
Yep, the logic is exactly the same, so moved this to a common utility that resides at: `cudf.core.tools.datetimes._isin_datetimelike`